app/misc/git.ts

```diff
@@ -8,21 +9,223 @@ let fs = require("fs");
+    .then(function (refs) {
+    // Check to see that reference is not to tag or remote
+      for (let i =0; i<refs.length; i++) {
+        if (!refs[i].isTag() && !refs[i].isRemote()) {
+          repo.getReferenceCommit(refs[i])
+          // Find all commits from branch
+          .then(function (commit) {
+            let history = commit.history(Git.Revwalk.SORT.Time);
+            history.on("end", function (commits) {
+
+              // Store commit hash for every commit in branch
+              for (let i = 0; i < commits.length; i++) {
+                if (commitList.indexOf(commits[i].toString()) < 0) {
+                  commitList.push(commits[i].toString());
+                }
+              }
+              // Prune early commits that do not belong to node
+              if (commitList.indexOf(beginningHash) >= 0) {
+                let stop = commitList.length - commitList.indexOf(beginningHash) -1;
+                console.log(stop);
+                commitList.splice(commitList.indexOf(beginningHash)+1, stop);
+              }
+              
+              console.log('number of commits');
+              console.log(numCommit);
+
+              // Prune later commits that do not belong to node. If no numCommit exist, prune list of commits down to 1
+              commitList.reverse();
+              if (numCommit > 0) {
+                let deleteNum = commitList.length - numCommit;
+                if (deleteNum > 0) {
+                  commitList.splice(numCommit, deleteNum);
+                }
+              }
+              // if (numCommit === '') {
+              //   if (commitList.length > 1) {
+              //     console.log(beginningHash);
+              //     let stop1 = commitList.length - commitList.indexOf(beginningHash) - 1;
+              //     commitList.splice(commitList.indexOf(beginningHash)+1, stop1);
+              //     console.log(commitList, stop1);
+              //     let stop2 = commitList.indexOf(beginningHash);
+              //     commitList.splice(0, stop2);
+              //     console.log(commitList);
+              //   } 
+              // } else if (commitList.indexOf(numCommit) >= 0) {
+              //   let stop = commitList.length - commitList.indexOf(numCommit) -1;
+              //   commitList.splice(commitList.indexOf(numCommit)+1, stop);
+              // } 
+              // console.log('Second Please');
+              // console.log(commitList);
+              // for (let k = 0; k < commitList.length; k++) {
+              //   repo.getCommit(commitList[k])
+              //     .then(function(commit1){
+              //       commitListRet.push(commit1);
+              //     })
+              // }
+              resolve(commitList);
+            });
+
+            history.start();
+          }).catch ((err) => {
+            console.log(err);
+          });
+        } else {
+          console.log('current branch does not exist on remote');
```

Comment:
Why are we console.logging? how often does this happen? should the user be notified or are you throwing an error?